tests/extract/test_decorators.py

```diff
@@ -853,3 +854,51 @@ def __call__(self, more: int = 1):
 @pytest.mark.skip("Not implemented")
 def test_class_resource() -> None:
     pass
+
+
+def test_parallelized_resource_decorator() -> None:
+    """Test paralellized resources are wrapped correctly.
+    Note: tests for parallel execution are in test_resource_evaluation
+    """
+
+    def some_gen():
+        yield from [1, 2, 3]
+
+    # Create resource with decorated function
+    resource = dlt.resource(some_gen, parallelized=True)
```

Comment:
also try `resource = dlt.resource(some_gen(), parallelized=True)` which will pass generator, not generator function.

we need separate test that will actually run such resource. you'll see that it will fail